okhttp/src/jvmMain/kotlin/okhttp3/internal/http/RealInterceptorChain.kt

```diff
@@ -82,6 +83,11 @@ class RealInterceptorChain(
     return copy(writeTimeoutMillis = checkDuration("writeTimeout", timeout.toLong(), unit))
   }
 
+  override fun withEventListener(eventListener: EventListener): Interceptor.Chain {
+    call.eventListener.addListener(eventListener)
```

Comment:
Does this need synchronization?

I thin maybe just documenting that this can only be called during the Interceptor.intercept when nothing else should be going on.

Reply:
I guess it does for cancellation.